File: FishBunDemo/src/main/java/com/sangcomz/fishbundemo/MainActivity.java
package com.sangcomz.fishbundemo;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    Button btnWithActivityLight;
    Button btnWithActivityBase;
    Button btnWithActivityDark;
    Button btnWithFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnWithActivityBase = (Button) findViewById(R.id.btn_with_activity_basic);
        btnWithActivityDark = (Button) findViewById(R.id.btn_with_activity_dark);
        btnWithActivityLight = (Button) findViewById(R.id.btn_with_activity_light);
        btnWithFragment = (Button) findViewById(R.id.btn_with_Fragment);

        btnWithActivityBase.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(MainActivity.this, WithActivityActivity.class);
                i.putExtra("mode", 0);
                startActivity(i);
            }
        });

        btnWithActivityDark.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(MainActivity.this, WithActivityActivity.class);
                i.putExtra("mode", 1);
                startActivity(i);
            }
        });

        btnWithActivityLight.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(MainActivity.this, WithActivityActivity.class);
                i.putExtra("mode", 2);
                startActivity(i);
            }
        });

        btnWithFragment.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(MainActivity.this, WithFragmentActivity.class);
                startActivity(i);
            }
        });
    }
}

File: FishBunDemo/src/main/java/com/sangcomz/fishbundemo/MainActivity.kt
package com.sangcomz.fishbundemo

import android.content.Intent
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import com.sangcomz.fishbundemo.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        enableEdgeToEdge()
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setupStatusBarInsets(binding.toolbar)
        setSupportActionBar(binding.toolbar)

        binding.btnWithActivityBasic.setOnClickListener {
            val i = Intent(this@MainActivity, WithActivityActivity::class.java)
            i.putExtra("mode", 0)
            startActivity(i)
        }

        binding.btnWithActivityDark.setOnClickListener {
            val i = Intent(this@MainActivity, WithActivityActivity::class.java)
            i.putExtra("mode", 1)
            startActivity(i)
        }

        binding.btnWithActivityLight.setOnClickListener {
            val i = Intent(this@MainActivity, WithActivityActivity::class.java)
            i.putExtra("mode", 2)
            startActivity(i)
        }

        binding.btnWithFragment.setOnClickListener {
            val i = Intent(this@MainActivity, WithFragmentActivity::class.java)
            startActivity(i)
        }
    }
}
